test/hotspot/gtest/opto/test_constant_division.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "opto/divconstants.hpp"
+#include "runtime/os.hpp"
+#include "utilities/growableArray.hpp"
+#include <type_traits>
```

Comment:
Move include.

Reply:
I was told that `unittest.hpp` should come last so this is the order, I have added a line between JDK header and stdlib header as well as resolved your other comments. Thanks a lot.

Reply:
The rules around the includes lines in our tests and what we currently have in the tests are messy at the movement. We should fix that when we find the time to.

For HotSpot source code files the includes should be structured as::
```
hotspot includes
blank line
system includes
```

There are some deviations from that, but those should be cleaned up instead of used as a precedence. For our tests we should add "unittest.hpp" at the end of the "hotspot includes" section.

Reply:
In the Oracle-internal discussion of include order from about a year ago, there was not a consensus
decision about the position of "unittest.hpp".  There was a concern that in some cases it really was
required to be last for some technical reason.  That needed (and still needs) investigation.

Reply:
I assume this means that the include order is good as it is now?

Reply:
Please update it to:
```
#include "precompiled.hpp"
#include "opto/divconstants.hpp"
#include "runtime/os.hpp"
#include "utilities/growableArray.hpp"
#include "unittest.hpp"

#include <type_traits>
```

Reply:
Got it, fixed.